### ISSUE TITLE
Functional Testing

### ISSUE THREAD

**Issue by dstreett** [dstreett/Super-Deduper#21]
Hey @bioSandMan,

I just merge your test into develop ran it. A little later I realized I was zeroing out the ID key to find duplicates so everything was considered a duplicate.

The 'make test' was OK in both. cases.

Example:
Building with correct source code. R1 and R2 looks as follows.

```

@NODUP
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+
BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB
@NODUP
TTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTT
+
BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB

```

Output of Super-Deduper

```
NODUP
TTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTT
+
BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB
@NODUP
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+
BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB

```

Totally expected to pass test and it does.

```
if [ ! -d "superTests" ]; then mkdir superTests; fi;
python testSD.py 
.
----------------------------------------------------------------------
Ran 1 test in 0.001s

OK
```

However, under the bug version. . .

The output with the same input is.

```

@NODUP
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
+
BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB


```

The other NODUP header doesn't show up.

The 'make test' shows this though.

```
if [ ! -d "superTests" ]; then mkdir superTests; fi;
python testSD.py 
.
----------------------------------------------------------------------
Ran 1 test in 0.001s

OK
```

Make test should fail under these conditions, but they don't seem to be. Let me know if you have any questions and when you would like me to test it again.

**Comment by bioSandMan** [comment]
At the moment everything will pass since I've been rebasing the tests on assertions. I should have this figured out shortly.

**Comment by dstreett** [comment]
Fixed with #23 PR.
